ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -270,30 +283,35 @@ def take_step(self, env, curr_info: AllBrainInfo):
             take_action_text[brain_name] = action_info.text
             take_action_value[brain_name] = action_info.value
             take_action_outputs[brain_name] = action_info.outputs
+        time_start_step = time()
         new_info = env.step(
             vector_action=take_action_vector,
             memory=take_action_memories,
             text_action=take_action_text,
             value=take_action_value
         )
-
+        delta_time_step = time() - time_start_step
         for brain_name, trainer in self.trainers.items():
+            if brain_name in self.trainer_metrics:
+                self.trainer_metrics[brain_name].add_delta_step(delta_time_step)
             trainer.add_experiences(curr_info, new_info,
                                     take_action_outputs[brain_name])
             trainer.process_experiences(curr_info, new_info)
             if trainer.is_ready_update() and self.train_model \
                     and trainer.get_step <= trainer.get_max_steps:
                 # Perform gradient descent with experience buffer
+
                 trainer.update_policy()
             # Write training statistics to Tensorboard.
+            delta_train_start = time() - self.training_start_time
             if self.meta_curriculum is not None:
                 trainer.write_summary(
                     self.global_step,
-                    lesson_num=self.meta_curriculum
+                    delta_train_start, lesson_num=self.meta_curriculum
                         .brains_to_curriculums[brain_name]
                         .lesson_num)
             else:
-                trainer.write_summary(self.global_step)
+                trainer.write_summary(self.global_step, delta_train_start)
```

Comment:
What about passing delta_time_step to this guy ? Instead of creating `add_delta_step`

Reply:
Overloads responsibility? I can see a situation where we can't avoid having this in `TrainerController`. I want to eliminate `add_delta_step` at some point post a cleaner refactor?